divideconquer/BinarySearch.java

```diff
@@ -0,0 +1,68 @@
+import java.io.*;
+import java.util.*;
+
+public class BinarySearch {
+
+    static int binarySearch(int[] a, int x, int left, int right) {
+        int mid = (right-left)/2;
+        if(left<right)
```

Comment:
Add whitespace to make your code more readable
`int mid = (right-left)/2;`
` `
`if(left<right)`